src/screens/atomic-swap/NewSwap.js

```diff
@@ -30,23 +39,68 @@ export default function NewSwap (props) {
 
     const createClickHandler = () => {
         if (password.length < 3) {
-            setErrorMessage('Please insert a password more than 3 characters long');
+            setErrorMessage(t`Please insert a password more than 3 characters long`);
             return;
         }
 
         setIsLoading(true);
-        const { data, id } = generateEmptyProposalFromPassword(password, wallet);
-        setProposalId(id);
-        dispatch(proposalGenerated(id, password, data))
+        const newPartialTx = generateEmptyProposal(wallet);
+        setPartialTx(newPartialTx)
+        dispatch(proposalCreateRequested(newPartialTx, password));
     }
 
+    /**
+     * Listening to the new proposal's status
+     */
+    useEffect(() => {
+        const creationStatus = newProposal?.status;
+
+        // Error handling
+        if (creationStatus === PROPOSAL_CREATION_STATUS.FAILED) {
+            setErrorMessage(newProposal.errorMessage)
+            setIsLoading(false);
+            return;
+        }
+
+        // While the proposal is not successfully created, just stay in this screen
+        if (creationStatus !== PROPOSAL_CREATION_STATUS.SUCCESS) {
+            return;
+        }
+
+        // If the proposal was successfully created, calculate its helper values and request temporary data cleanup
+        const proposalId = newProposal.proposalId;
+        const reduxObj = generateReduxObjFromProposal(
+          proposalId,
+          password,
+          partialTx,
+          wallet,
+        );
+        reduxObj.status = PROPOSAL_DOWNLOAD_STATUS.READY; // We already have all data, no need to fetch again.
+
+        // Adding the new redux object to the listened proposals map
+        dispatch(proposalCreateCleanup(reduxObj));
+        setNewProposalId(proposalId);
+    }, [newProposal])
+
+    /**
+     * Listening to the proposals map when the cleanup was requested
+     */
     useEffect(() => {
```

Comment:
Why do you need this second `useEffect`?

Can't you call 

```
            updatePersistentStorage(allProposals);
            navigateToProposal(proposalId);
```

In the first one after updating the proposals map in redux?

Reply:
There are two saga flows that happen for the creation of a swap:
- The first generates the identifier on the backend
- The second adds the proposal data to the listened proposals map

Added some more comments to this code for clarity.